lib/pure/unicode.nim

```diff
@@ -1602,6 +1605,25 @@ iterator runes*(s: string): Rune =
     fastRuneAt(s, i, result, true)
     yield result
 
+when not defined(nimscript):
+  iterator runes*(s: Stream): Rune =
+    ## Iterates over any unicode character of
+    ## the stream ``s`` returning runes. Raises `EIO`
+    ## if an error occurred.
+    var buff = newString(4).TaintedString
+    var rune: Rune
+    var n = 0
+    while true:
+      let l = peekData(s, addr(string(buff)[0]), string(buff).len)
```

Comment:
Peeking is not always possible, better read and see whether you reached the end.

Reply:
But then this would consume more than it should. There would be no way to "read until finding X" then do something else with the stream.